Elements/INavigableExtensions.cs

```diff
@@ -0,0 +1,108 @@
+﻿using System.Diagnostics.CodeAnalysis;
+using UnityEngine.UI;
+
+namespace Silksong.ModMenu.Elements;
+
+/// <summary>
+/// Helper functions for INavigable instances.
+/// </summary>
+public static class INavigableExtensions
+{
+    /// <summary>
+    /// Helper functions for INavigable instances.
+    /// </summary>
+    extension(INavigable self)
+    {
+        /// <summary>
+        /// Set the upwards neighbor of this navigable.
+        /// </summary>
+        public void SetNeighborUp(Selectable selectable) =>
+            self.SetNeighbor(NavigationDirection.Up, selectable);
+
+        /// <summary>
+        /// Set the upwards neighbor of this navigable.
+        /// </summary>
+        public void SetNeighborUp(SelectableElement selectableElement) =>
+            self.SetNeighborUp(selectableElement.SelectableComponent);
+
+        /// <summary>
+        /// Declare that this navigable has no upwards neighbor.
+        /// </summary>
+        public void ClearNeighborUp() => self.ClearNeighbor(NavigationDirection.Up);
+
+        /// <summary>
+        /// Get the most eligible selectable within this navigable when navigating upwards into it.
+        /// </summary>
+        public bool GetNeighborUp([MaybeNullWhen(false)] out Selectable selectable) =>
```

Comment:
I usually see this pattern represented as a nullable with `NotNullWhen(true)` but I don't think it's semantically different